cadence/contracts/connectors/increment-fi/IncrementFiStakingConnectors.cdc

```diff
@@ -150,20 +148,22 @@ access(all) contract IncrementFiStakingConnectors {
         /// @param userCertificate: Capability to access the user's staking certificate
         /// @param poolID: The unique identifier of the staking pool to claim rewards from
         /// @param vaultType: The type of Vault this Source provides when claiming rewards
-        /// @param overflowSinks: A set of DeFiActions.Sink to handle any overflow from the rewards claim
         /// @param uniqueID: Optional identifier for associating connectors in a stack
         ///
         init(
             userCertificate: Capability<&Staking.UserCertificate>,
             poolID: UInt64,
-            vaultType: Type,
-            overflowSinks: {Type: {DeFiActions.Sink}},
             uniqueID: DeFiActions.UniqueIdentifier?,
         ) {
+            let pool = IncrementFiStakingConnectors.borrowPool(poolID: poolID) ?? panic("Pool with ID \(poolID) not found")
+            let rewardsInfo = pool.getPoolInfo().rewardsInfo
+
+            assert(rewardsInfo.keys.length == 1, message: "Pool with ID \(poolID) has multiple reward token types, only one is supported")
```

Comment:
Is this a safe assumption to make? It looks like the type is a mapping, but is it relatively uncommon for the rewards to contain more than one Vault type?